src/main/starlark/builtins_bzl/common/objc/compilation_support.bzl

```diff
@@ -42,6 +42,31 @@ def _build_variable_extensions(ctx, arc_enabled):
 
     return extensions
 
+def _create_compilation_attributes(ctx):
+    disallow_sdk_frameworks = ctx.fragments.objc.disallow_sdk_frameworks_attributes
+    sdk_frameworks = getattr(ctx.attr, "sdk_frameworks", [])
+    weak_sdk_frameworks = getattr(ctx.attr, "weak_sdk_frameworks", [])
+    if disallow_sdk_frameworks:
+        if sdk_frameworks:
+            fail("sdk_frameworks attribute is disallowed. Use explicit dependencies instead.")
+        if weak_sdk_frameworks:
+            fail("weak_sdk_frameworks attribute is disallowed. Use explicit dependencies instead.")
+
+    return struct(
+        hdrs = depset(cc_helper.get_public_hdrs(ctx)),
```

Comment:
Idk if it matters if these are depsets anymore but the callers assumed they were. I can change that if folks think we should

Reply:
Bazel maintains a cache for depsets that just wrap an `ImmutableList`, so this may end up saving retained memory even though it looks like a waste at first. Ran into this a few times while implementing path mapping for C++ rules.